File: integrasjon/src/main/java/no/nav/melosys/integrasjon/ereg/EregFasade.java
package no.nav.melosys.integrasjon.ereg;

import java.util.Optional;

import no.nav.melosys.domain.Saksopplysning;
import no.nav.melosys.exception.IkkeFunnetException;

/**
 * Fasade mot Enhetsregisteret (EREG)
 */
public interface EregFasade {

    Saksopplysning hentOrganisasjon(String orgnr) throws IkkeFunnetException;

    Optional<Saksopplysning> finnOrganisasjon(String orgnr);

    String hentOrganisasjonNavn(String orgnummer) throws IkkeFunnetException;
}

File: integrasjon/src/main/kotlin/no/nav/melosys/integrasjon/ereg/EregFasade.kt
package no.nav.melosys.integrasjon.ereg

import no.nav.melosys.domain.Saksopplysning
import no.nav.melosys.exception.IkkeFunnetException
import java.util.*

/**
 * Fasade mot Enhetsregisteret (EREG)
 */
interface EregFasade {

    @Throws(IkkeFunnetException::class)
    fun hentOrganisasjon(orgnr: String): Saksopplysning

    fun finnOrganisasjon(orgnr: String): Optional<Saksopplysning>

    @Throws(IkkeFunnetException::class)
    fun hentOrganisasjonNavn(orgnummer: String): String
}
